ignite/metrics/accumulation.py

```diff
@@ -38,7 +38,10 @@ class VariableAccumulation(Metric):
     _required_output_keys = None
 
     def __init__(
-        self, op: Callable, output_transform: Callable = lambda x: x, device: Optional[Union[str, torch.device]] = None
+        self,
+        op: Callable,
+        output_transform: Callable = lambda x: x,
+        device: Optional[Union[str, torch.device]] = torch.device("cpu"),
```

Comment:
I wonder if `Optional` is still makes sense here now. Device possible values now are "cpu", "cuda", ... and same with `torch.device`... If we put None it is the same as CPU. I think if we keep None as option, than default value should be None too. What do you think ?

Reply:
Makes total sense, thanks for catching this. I agree that this should just be `Union[str, torch.device]`. I'll make that change.

Reply:
@n2cholas actually, why we needed this change ? When None it is same as torch device CPU ?

Reply:
@vfdev-5 Calling `tensor.to(None)` acts like the identity: it will keep the tensor on whatever device it was already on. I needed to make this change to fix the TPU tests. In some tests, the metric had no specified device but the inputs to `update` were on the TPU, so when we tried to add values to the accumulator we had a device mismatch.

Reply:
Okay, I see. Thanks ! Explicit device values is better then None reassigned afterward.